### PR TITLE
ci: Add Conventional Commit Lint

### PR DESCRIPTION
**Pull request by arielvalentin**
This is an attempt to add conventional commit lint.

It is an alternative to https://github.com/open-telemetry/opentelemetry-ruby-contrib/pull/800

### CODE REVIEW

**Comment by arielvalentin** [comment]
@scbjans @kaylareopelle I got something working but it required two different GitHub actions to get what I wanted. 

1. Action checks the PR title
2. Action that checks individual commit messages

**Comment by arielvalentin** [comment]
yup... seems like merge commits are ok https://github.com/open-telemetry/opentelemetry-ruby-contrib/actions/runs/7551640637/job/20559186184?pr=821

File: .github/workflows/lint-conventional-commits.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+          types: |
+            fix
+            feat
+            chore
+            ci
+            docs
+            style
+            refactor
+            perf
+            test
```
Comment (kaylareopelle) [inline .github/workflows/lint-conventional-commits.yaml:36]:
```suggestion
            chore
            ci
            docs
            feat
            fix
            perf
            refactor
            style
            test
```
Mind if we alphabetize this list?

Also, the list in `commitlint.config.js` also includes `revert` and `squash`. Should those be here too?

**Review by kaylareopelle — COMMENTED** [review]
@arielvalentin - Yay! I'm so glad it's working. I have two questions.

**Comment by scbjans** [comment]
> yup... seems like merge commits are ok https://github.com/open-telemetry/opentelemetry-ruby-contrib/actions/runs/7551640637/job/20559186184?pr=821

Yoohoo! Is there a limit on the number of GH actions we can use? 
If so https://github.com/open-telemetry/opentelemetry-ruby-contrib/pull/800 seems to be unblocked now; so that would be more favorable then

**Comment by arielvalentin** [comment]
closed in favor of #800

**Pull request closed without merging**